src/clients/docs_generate.zig

```diff
@@ -0,0 +1,107 @@
+        try writer.print("#", .{});
+        x -= 1;
+    }
+
+    try writer.print(" {s}\n\n", .{content});
+}
+
+pub fn printParagraph(writer: std.fs.File.Writer, content: []const u8) !void {
+    try writer.print("{s}\n\n", .{content});
+}
+
+pub fn printCodeBlock(writer: std.fs.File.Writer, language: []const u8, content: []const u8) !void {
+    try writer.print("```{s}\n{s}\n```\n\n", .{ language, content });
+}
+
+pub fn printLanguageCodeBlock(writer: std.fs.File.Writer, language: Docs, content: []const u8) !void {
+    try printCodeBlock(writer, language.markdown_name, content);
+}
+
+pub fn main() !void {
+    for (languages) |language| {
+        //try validate_docs(language);
+
+        const file = try std.fs.cwd().openFileZ(language.readme, .{ .mode = .write_only });
+        // Truncate file
+        try file.setEndPos(0);
+        defer file.close();
+
+        const writer = file.writer();
+
+        try printHeader(writer, 1, language.name);
+        try printParagraph(writer, language.description);
+
+        try printHeader(writer, 2, "Installation");
+        try printCodeBlock(writer, "bash", language.install_commands);
+
+        if (language.examples.len != 0) {
+            try printHeader(writer, 2, "Examples");
+            try printParagraph(writer, language.examples);
+        }
+
+        try printHeader(writer, 2, "Creating a Client");
+        try printLanguageCodeBlock(writer, language, language.client_object_example);
+        try printParagraph(writer, language.client_object_documentation);
+
+        try writer.print("The following are valid addresses:\n", .{});
+        try writer.print("* `3000` (interpreted as `127.0.0.1:3000`)\n", .{});
+        try writer.print("* `127.0.0.1:3000` (interpreted as `127.0.0.1:3000`)\n", .{});
+        try writer.print("* `127.0.0.1` (interpreted as `127.0.0.1:3001`, `3001` is the default port)\n\n", .{});
+
+        try printHeader(writer, 2, "Creating Accounts");
+        try printParagraph(writer, "See details for account fields in the [Accounts reference](https://docs.tigerbeetle.com/reference/accounts).");
+        try printLanguageCodeBlock(writer, language, language.create_accounts_example);
+        try printParagraph(writer, language.create_accounts_documentation);
+
+        try printHeader(writer, 3, "Account Flags");
+        try printParagraph(writer, "The account flags value is a bitfield. See details for these flags in the [Accounts reference](https://docs.tigerbeetle.com/reference/accounts#flags).");
+        try printParagraph(writer, language.account_flags_details);
+
+        try printHeader(writer, 3, "Response and Errors");
+        try printParagraph(writer, "The response is an empty array if all accounts were created successfully. If the response is non-empty, each object in the response array contains error information for an account that failed. The error object contains an error code and the index of the account in the request batch.");
```

Comment:
do we want to wrap these long lines?

Reply:
I personally find the `\\` syntax quite nice, not messing up indentation is a big win.